parlai/utils/conversations.py

```diff
@@ -93,6 +95,74 @@ def save_metadata(
             f.write(json.dumps(metadata))
 
 
+class Turn(AttrDict):
+    """
+    Utility class for a dialog turn.
+    """
+
+    def __init__(self, id=None, text=None, **kwargs):
+        super().__init__(self, id=id, text=text, **kwargs)
+
+
+class Conversation:
+    """
+    Utility class for iterating through a single episode.
+
+    Used in the context of the Conversations class.
+    """
+
+    def __init__(self, episode):
+        self.episode = episode
+        self.context = episode.get('context')
+        self.metadata_path = episode.get('metadata_path')
+        self.turns = self._build_turns(episode)
+        self.iterator_idx = 0
+
+    def _build_turns(self, episode):
+        turns = []
+        for act_pair in episode['dialog']:
+            for act in act_pair:
+                turns.append(Turn(**act))
+        return turns
+
+    def read(self):
```

Comment:
I agree with Kurt, It would be nice to have a `__str__` that one would call print.

a `__repr__` would also be nice, but it could be more high level like `Conversation(turns=5, first="Hi how are, ...")` or something like that?